ARCHITECTURE.md

```diff
@@ -1,113 +1,265 @@
+In order to allow the users to reconnect to the game and to be able to restore their game state - we are keeping a map of their GUIDs to their respective data. This way we ensure that when a player disconnects, some data is kept and accurately assigned back to that player when they reconnect.
 
-**Server**
- - `ServerCharacterMovement` manages the movement Finite State Machine (FSM) on the server. Updates the NetworkVariables that synchronize position, rotation and movement speed of the entity on its FixedUpdate.
- - `ServerCharacter` has the `AIBrain`, as well as the ActionQueue. Receives action requests (either from the AIBrain in case of NPCs, or user input in case of player characters), and executes them.
- - `AIBrain` contains main AI FSM.  
- - `Action` is the abstract base class for all server actions
-   - `MeleeAction`, `AoeAction`, etc. contain logic for their respective action types. 
+For more information check out the page on [Session Management](https://docs-multiplayer.unity3d.com/netcode/current/advanced-topics/session-management/index.html) in NGO documentation.
 
-**Client**
- - `ClientCharacterVisualization` primarily is a host for the running `ActionFX` class.
- - `ClientInputSender `. On a shadow entity, will self-destruct. Listens to inputs, interprets them, and then calls appropriate RPCs on the RPCStateComponent. 
- - `ActionFX` is the abstract base class for all the client-side action visualizers
-   - `MeleeActionFX`, `AoeActionFX`, etc. Contain graphics information for their respective action types. 
-  
-## Movement action flow
- - Client clicks mouse on target destination. 
- - Client->server RPC, containing target destination. 
- - Anticipatory animation plays immediately on client. 
- - Server performs pathfinding.
- - Once pathfinding is finished, server representation of entity starts updating its NetworkVariables at the same cadence as FixedUpdate.
- - Visuals GameObject never outpaces the simulation GameObject, and so is always slightly behind and interpolating towards the networked position and rotation.
+## UGS Services integration - Lobby and Relay
 
-## Navigation System
-Each scene which uses navigation or dynamic navigation objects should have a `NavigationSystem` component on a scene GameObject. That object also needs to have the `NavigationSystem` tag.
+Boss Room is a multiplayer experience that’s meant to be playable over internet, and this is why we’ve integrated [Unity Gaming Services](https://unity.com/solutions/gaming-services) – a combination of Authentication, Lobby, and Relay is what allows us to make it easy for players to host and join games that are playable over the internet, without the need for port forwarding or out-of-game coordination.
+
+The following classes are of interest to learn more about our UGS wrappers and integration:
```

Comment:
It'd be great to have a line or two on Facade and Mediator and why we have this structure.